flax/experimental/nnx/nnx/graph.py

```diff
@@ -913,6 +913,30 @@ def update(
     )
     return unflatten(new_graphdef, state, idxmap=index_to_ref)[0]
 
+  def clear_refmap(self):
+    if self.refmap is None:
+      raise ValueError('Cannot clear refs without refmap.')
```

Comment:
`RuntimeError`? I assume you don't really expect the caller to handle this exception.